cpp/example/run.sh

```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/env bash
 
-ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE:-$0}")" || exit; pwd)"
-
 bazel --nosystem_rc --nohome_rc build //:example
 if [[ "$OSTYPE" == "darwin"* ]]; then
-    DYLD_LIBRARY_PATH="$ROOT_DIR/thirdparty/lib" "${ROOT_DIR}"/bazel-bin/example
+    install_name_tool -change bazel-out/darwin-opt/bin/cpp/libray_api.so ./thirdparty/lib/libray_api.so ./bazel-bin/example.so
+    install_name_tool -change bazel-out/darwin-opt/bin/cpp/libray_api.so ./thirdparty/lib/libray_api.so ./bazel-bin/example
```

Comment:
Can we use install_name_tool when we copy `libray_api.so` to `thirdparty` directory? This seems to be a little weird to `install_name_tool ` in users' script.

Reply:
We need to modify the dependency so path of example and example.so, modify libray_api.so not work.